bftengine/src/bftengine/ReplicaBase.hpp

```diff
@@ -80,11 +81,17 @@ class ReplicaBase {
 
   void sendRaw(MessageBase* m, NodeIdType dest);
 
-  bool validateMessage(MessageBase* msg) {
+  template <typename MessageType>
+  bool validateMessage(MessageType* msg) {
+    if (config_.debugStatisticsEnabled) {
+      DebugStatistics::onReceivedExMessage(msg->type());
+    }
     try {
-      if (config_.debugStatisticsEnabled) DebugStatistics::onReceivedExMessage(msg->type());
-
-      msg->validate(*repsInfo);
+      if constexpr (std::is_same_v<MessageType, CheckpointMsg>) {
```

Comment:
It is a good idea to document why the checkpoint message is handled differently.

Reply:
done